src/trio/_core/_tests/test_guest_mode.py

```diff
@@ -664,3 +665,51 @@ async def trio_main() -> None:
     context.run(aiotrio_run, trio_main, host_uses_signal_set_wakeup_fd=True)
 
     assert record == {("asyncio", "asyncio"), ("trio", "trio")}
+
+
+@restore_unraisablehook()
+def test_guest_mode_asyncgens_garbage_collection() -> None:
```

Comment:
I had a few comments about test code but seeing as you copied the test from above, I guess I should put them in one place.

Firstly, I don't think it's necessary to check the library labels.

Then specifics (which I guess also apply to the previous test function too):
 - the 3.6 compat code is unnecessary
 - I assume the `import sniffio` can be lifted to the top level? (Probably not even necessary for this test because we shouldn't check labels)
 - is the unraisable hook even necessary? I don't see where we could be raising in a garbage collection. (IIRC that's what the decorator is for?)

Reply:
This is disabling the default pytest unreasonable hook which would fail the test when the async generator is finalized because trio warns about it

Reply:
I'd like to keep the labels just so it's clear which generator gets finalized when

Reply:
Yeah that's reasonable. `== {True}` isn't the most informative in telling the reader "this True happened twice, by the way".